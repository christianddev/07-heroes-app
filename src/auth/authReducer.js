import {types} from '../types/types';

// const statec = {
//     name: 'David',
//     logged: true
// };

const authReducer = (user = {}, action)=>  {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                logged: false
            }
        default:
            return user;
    }
}
export default authReducer;