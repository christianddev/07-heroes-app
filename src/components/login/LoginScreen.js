import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

const LoginScreen = ({history}) => {
  const {dispatch} = useContext(AuthContext);

  const handleLogin = () => {
    dispatch({
    type: types.login,
    payload: {
      name: 'David'
    }
  })

  const lastPath = localStorage.getItem('lastPath') || '/';

  history.replace(lastPath);

  };

  return (
    <div className="container mt-5">
      <h1>LoginScreen</h1>
      <hr/>
      <button
        onClick={handleLogin}
        className="btn btn-primary"
      >Login</button>
    </div>
  )
}

export default LoginScreen
