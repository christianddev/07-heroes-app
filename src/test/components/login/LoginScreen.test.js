import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";

import LoginScreen from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";
import { historyMock } from "../../fixtures/historyMock";

describe("LoginScreen.js", () => {
  const contextValue = {
    dispatch: jest.fn(),
  };
  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  );

  test("show LoginScreen", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("show call handleLogin", () => {
    const handleClick = wrapper.find('button').prop('onClick');
    const dispatch = {'payload': {'name': 'David'}, type: types.login}

    handleClick();
    expect(contextValue.dispatch).toHaveBeenCalledWith(dispatch);
    expect(historyMock.replace).toHaveBeenCalledWith('/');
    localStorage.setItem('lastPath', '/dc')
    handleClick();
    expect(historyMock.replace).toHaveBeenCalledWith('/dc');
  });
});
