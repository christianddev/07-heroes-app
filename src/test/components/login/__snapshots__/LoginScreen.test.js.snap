// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen.js show LoginScreen 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
